superagi/controllers/agent_template.py

```diff
@@ -209,44 +210,62 @@ def save_agent_as_template(agent_execution_id: str,
     Raises:
         HTTPException (status_code=404): If the agent or agent execution configurations are not found.
     """
+
     if agent_execution_id == 'undefined':
         raise HTTPException(status_code = 404, detail = "Agent Execution Id undefined")
-
-    agent_executions = AgentExecution.get_agent_execution_from_id(db.session, agent_execution_id)
-    if agent_executions is None:
-        raise HTTPException(status_code = 404, detail = "Agent Execution not found")
-    agent_id = agent_executions.agent_id
+    if agent_id == 'undefined':
+        raise HTTPException(status_code = 404, detail = "Agent Id undefined")
 
     agent = db.session.query(Agent).filter(Agent.id == agent_id).first()
     if agent is None:
         raise HTTPException(status_code=404, detail="Agent not found")
+    
+    #Fetch agent id from agent execution id and check whether the agent_id received is correct or not.
+    if agent_execution_id!="-1": 
+        agent_execution_config = AgentExecution.get_agent_execution_from_id(db.session, agent_execution_id)
+        if agent_execution_config is None:
+            raise HTTPException(status_code = 404, detail = "Agent Execution not found")
+        agent_id_from_execution_id = agent_execution_config.agent_id
+        if agent_id != agent_id_from_execution_id:
+            raise HTTPException(status_code = 404, detail = "Wrong agent id")
 
-    agent_execution_configurations = db.session.query(AgentExecutionConfiguration).filter(AgentExecutionConfiguration.agent_execution_id == agent_execution_id).all()
-    if not agent_execution_configurations:
-        raise HTTPException(status_code=404, detail="Agent configurations not found")
+    main_keys = AgentTemplate.main_keys()
```

Comment:
remove this variable instead directly use in call method, these variables make the method long and complex